client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -277,60 +278,64 @@ private ResponseOrResponseException convertResponse(InternalRequest request, Nod
      * @param responseListener the {@link ResponseListener} to notify when the
      *      request is completed or fails
      */
-    public void performRequestAsync(Request request, ResponseListener responseListener) {
+    public Cancellable performRequestAsync(Request request, ResponseListener responseListener) {
```

Comment:
Do we need a custom class ? Why can't we use a simple `Runnable` ?

Reply:
I don't follow how Runnable would work here. You want something to call cancel on, which in turns calls abort on the request. Users would call run against the runnable?